src/openrct2/world/Map.cpp

```diff
@@ -1983,41 +1983,41 @@ static void map_invalidate_tile_under_zoom(int32_t x, int32_t y, int32_t z0, int
  *
  *  rct2: 0x006EC847
  */
-void map_invalidate_tile(int32_t x, int32_t y, int32_t z0, int32_t z1)
+void map_invalidate_tile(const CoordsXY& tilePos, int32_t z0, int32_t z1)
```

Comment:
This function, and its siblings, receives CoordsXY, coupled with TileCoords z values. I want to get rid of this mixed kind of functions.

One thing, of course, is that we don't have a fitting coords type for functions that need both a low and high Z. I propose creating CoordsXYZZ for this purpose.

@duncanspumpkin

Reply:
I'm not sure cordsxyzz makes that much sense. It's not exactly used that often either.

Reply:
I think there are at least 20 functions that have this pattern. I really think such a struct should be created.

I'm not wedded to the CoordsXYZZ name, though. It just seemed a straightforward name to me.

Reply:
Maybe `CoordsXYRangedZ` or something alike

Reply:
@Gymnasiast thoughts on `CoordsXYRangedZ`? I think that is a bit better. I'm not overly fussed on what is chosen. We should perhaps make a `RangedZ` struct though as there are some instances I can think of (map is clear at functions) where you pass in a base and clearance.

Reply:
`CoordsXYRangedZ` is fine.

As for the `RangedZ` struct: are there places where one needs to pass _just_ a ranged Z and no X/Y?

Reply:
Yes the map is clear functions use it to work out collisions. There might be a few others

Reply:
Do you mean `map_can_construct_with_clear_at()`? That one also takes x and y.

I haven't been able to find a function that _just_ takes a ranged z (and no x/y).

Reply:
Oh yeah I'm a twit. Yes of course it does have x and y as well. Yes just create a new struct called `CoordsXYRangedZ` and this will cover all cases I think.

Reply:
Implemented and replaced all usages with `CoordsXYRangedZ`

Reply:
Good!

I think it's best done in a separate PR. I will review this one and merge it if nothing else pops up.